### PR TITLE
Add more utility procedures to small array

### PR DESCRIPTION
**Pull request by Lperlind**
Just some more procedures that I'm using in my project which I thought I should just put out there.

### CODE REVIEW

File: core/container/small_array/small_array.odin [hunk 1]
```diff
@@ -32,6 +32,20 @@ get_ptr :: proc "contextless" (a: ^$A/Small_Array($N, $T), index: int) -> ^T {
 	return &a.data[index]
 }
 
+get_safe :: proc(a: $A/Small_Array($N, $T), index: int) -> (T, bool) {
```
Comment (Skytrias) [inline core/container/small_array/small_array.odin:35]: looks good to me

File: core/container/small_array/small_array.odin [hunk 2]
```diff
@@ -102,6 +116,19 @@ consume :: proc "odin" (a: ^$A/Small_Array($N, $T), count: int, loc := #caller_l
 	a.len -= count
 }
 
+ordered_remove :: proc "odin" (a: ^$A/Small_Array($N, $T), index: int, loc := #caller_location) {
```
Comment (Skytrias) [inline core/container/small_array/small_array.odin:119]: could maybe do what the builtin `order_remove` does -> https://github.com/odin-lang/Odin/blob/master/core/runtime/core_builtin.odin#L61

maybe also use runtime.bounds_check_error_loc with `#no_bounds_check` flag

File: core/container/small_array/small_array.odin [hunk 3]
```diff
@@ -111,6 +138,18 @@ push_back_elems :: proc "contextless" (a: ^$A/Small_Array($N, $T), items: ..T) {
 	a.len += n
 }
 
+insert :: proc "contextless" (a: ^$A/Small_Array($N, $T), item: T, index: int) -> bool {
```
Comment (Skytrias) [inline core/container/small_array/small_array.odin:141]: could be renamed to `inject_at` to keep the naming scheme consistent with core

**Comment by Lperlind** [comment]
Added tests and addressed comments

**Pull request merged**